packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -165,11 +165,12 @@ function Tab<T>(props: TabProps<T>) {
     ...props
   });
   let isSelected = state.selectedKey === key;
+  let domProps = filterDOMProps(item.props);
```

Comment:
For safety, mind extracting `id` from `domProps` here so that it doesn't make it to the tab div? We generate a specific id for the Tab in `useTab` that we don't wanna overwrite if the user provides an id

Reply:
Added deletion and test.